### PR TITLE
core refactor

### PR DESCRIPTION
**Pull request by z2sx**
Signed-off-by: Andrey Platov <andrey.v.platov@gmail.com>

### CODE REVIEW

File: packages/core/src/core.ts [hunk 1]
```diff
@@ -114,3 +115,21 @@ export const CORE_MIXIN_INDICES = 'mixin:core.Indices' as Ref<Mixin<Indices>>
 export interface AnyLayout {
   [key: string]: PropertyType
 }
+
+//
+
+export interface Tx extends Doc {
+  _date: DateProperty
```
Comment (haiodo) [inline packages/core/src/core.ts:122]: Is it still relevant to use '_'?

Comment (z2sx) [reply packages/core/src/core.ts:122]: no :) I just preserved it for "compatibility", afraid to break more things if remove '_'

Comment (z2sx) [reply packages/core/src/core.ts:122]: maybe later I'll do cleanup job to remove '_'s everywhere without introducing other changes 😄

**Review by haiodo — APPROVED** [review]
Looks good, some cleanups are required, but it is ok.

**Pull request merged**